PyInstaller/building/datastruct.py

```diff
@@ -173,6 +178,17 @@ def __postinit__(self):
             else:
                 # create a dict for easier access
                 data = dict(zip((g[0] for g in self._GUTS), data))
+
+                # this ::HACK:: converts winmanifest from a string
+                if is_win and data.get('manifest'):
+                    if not data['manifest'].startswith(winmanifest_marker):
```

Comment:
To follow up on the possible changes to EXE.  I looked at moving the manifest create to assemble(), but there was some code between \__init__() and assemble() that expected the manifest object to exist.  Since this code is generally only run once per build I decided not persue it any farther.